yellowbrick/base.py

```diff
@@ -206,7 +209,14 @@ def poof(self, outpath=None, clear_figure=False, **kwargs):
         primarily called by the user to render the visualization.
         """
         # Ensure that draw has been called
-        if self._ax is None: return
+        if self._ax is None:
+            warn_message = (
+                "{} does not have a reference to a matplotlib.Axes "
+                "the figure may not render as expected!"
+            )
+            warnings.warn(
+                warn_message.format(self.__class__.__name__), YellowbrickWarning
+            )
```

Comment:
This is part of the JointPlot fix for #716 -- the `self._ax is None` check is to ensure that we don't poof and therefore lock a figure before draw is called. However, having it return silently meant that it was hard to figure out the joint plot bug. 

Right now I'm issuing a warning and continuing but I don't know if this is the best idea. 

Other options:

1. Raise an exception
2. Issue a warning and return 
3. Drop the check 

If we think this discussion is too big for this PR we can make it another issue.

Reply:
I am also unsure what the best course of action would be

Reply:
Ok, sounds good - let's all just be aware that if we see this warning reported in an issue or as we are working with YB, we should take action and create a new issue for it. There are two possibilities if this happens:

- user error, calling `poof()` without calling `fit()`
- developer/api error, as in joint plot 

I'm worried about the latter because I can't remember why we put this check in originally.